Entities/Industry/Fireplace/Fireplace.as

```diff
@@ -98,27 +106,18 @@ void Extinguish(CBlob@ this)
 
 void Ignite(CBlob@ this)
 {
-	if (this.getSprite().isAnimation("fire")) return;
-
-	this.SetLight(true);
-	this.Tag("fire source");
-
-	this.getSprite().SetAnimation("fire");
-	this.getSprite().SetEmitSoundPaused(false);
-	this.getSprite().PlaySound("/FireFwoosh.ogg");
-	
-	CSpriteLayer@ fire = this.getSprite().getSpriteLayer("fire_animation_large");
-	if (fire !is null)
-	{
-		fire.SetVisible(true);
-	}
+	IgniteFireplace(this);
 }
```

Comment:
You don't need to wrap `IgniteFireplace` in another function. Just use `IgniteFireplace` directly.